woocommerce/woocommerce-android
# Added new site picker fragment + xml

Part 2 #6111. 

## Changes
- This PR adds a `SitePickerRepository` and a `SitePickerViewModel` to the app. 
- The repository is the only class that connects with FluxC and the REST API/Db and the ViewModel includes all the business logic for the site picker. 
- I added a new Site picker fragment that is added to the `MainActivity` nav graph, rather than using a separate activity.
- The subsequent PR will add the site picker UI + fragment changes. 
- The PR looks big but most of the files changes are `string.xml`. I removed a couple of strings from all the translated files to make lint pass.

## Logic
Looking at the `SitePickerActivity`, I noticed different flows and most of this logic is migrated to the ViewModel. I'll attempt to share the logic here in the hopes that the reviewer would have better clarity when looking at the code and can offer suggestions/improvements, if needed.
- The Site picker is called when the user first logs into the app AND when they would like to switch stores. The UI for both is different and based on the `openedFromLogin` nav args, we can differentiate between the two scenarios.
- If no sites are found for the user, we can assume that either Jetpack or Woo is not available on their site and ask them to install both before trying again.
- If user is logging in via the site address flow, we store the site address in `SharedPreferences` and initiate the site verification process automatically when the site picker is called. 
	- If the site is not found, then we can assume that the account the user has logged into does not match the site.
	- If the site does not have WooCommerce, then we can ask the merchant to install, activate Woo on their site.

### Testing instructions
- No user facing changes at the moment. Just run `SitePickerViewModelTest` and ensure everything passes.
